### PR TITLE
Omit .dice files in Bundler path or in dot-prefixed directories

### PR DESCRIPTION
**Pull request by harryw**
This adds an implicit feature to ignore .dice files that have been installed with vendored gems in a subdirectory.

Cucumber tests are included. Testing was by far the most difficult part of this change. The tests run Bundler to install dice_bag's dependencies in a subdirectory so they are slower than other tests, but still only a few seconds.

I've also added a test to document the existing behaviour with dot-prefixed directories, as described in issue #69.

### CODE REVIEW

File: features/ignore_configuration_files_in_bundled_gems.feature [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  Bundler may be used to install dependent gems that include their own
+  Dice Bag configuration files. It's also possible that these gems may
+  be installed in a local directory such as './vendor/bundle'. In these
+  circumstances, we should ignore any Dice Bag configuration files in
+  that local directory.
+
+  @clean-bundler-environment
+  Scenario: Ignoring files in the bundle path
+    Given The default aruba timeout is 15 seconds
```
Comment (jcarres-mdsol) [inline features/ignore_configuration_files_in_bundled_gems.feature:11]: How long it takes for the tests now? I am afraid of this being fragile.

**Comment by jcarres-mdsol** [comment]
+1 
@asmith-mdsol  I think it is good to merge

File: features/ignore_configuration_files_in_bundled_gems.feature (hunk 1 above)
Comment (harryw) [inline features/ignore_configuration_files_in_bundled_gems.feature:11]: The new tests add around 10 seconds on my laptop on a fairly standard household internet connection.

**Comment by benton** [comment]
I think so too. Passes all tests on my workstation, and seems to behave correctly when configuring Eureka. Nice work.

File: features/ignore_configuration_files_in_bundled_gems.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Feature: Ignore configuration files in bundled gems
+
+  Bundler may be used to install dependent gems that include their own
+  Dice Bag configuration files. It's also possible that these gems may
```
Comment (asmith-mdsol) [inline features/ignore_configuration_files_in_bundled_gems.feature:4]: If I were a horrible person I might make you correct the spelling to "DiceBag" here... but I'll settle for this passive-aggressive comment instead. :wink:

File: features/ignore_configuration_files_in_bundled_gems.feature (hunk 1 above)
Comment (asmith-mdsol) [inline features/ignore_configuration_files_in_bundled_gems.feature:11]: Is this a built-in Aruba step?

File: features/support/env.rb [hunk 3]
```diff
@@ -1,2 +1,3 @@
 require 'aruba/cucumber'
 require 'dice_bag'
+
```
Comment (asmith-mdsol) [inline features/support/env.rb:3]: Blank line?! Table-flip!

**Comment by asmith-mdsol** [comment]
Thanks for diligent testing here Harry!

File: features/ignore_configuration_files_in_bundled_gems.feature (hunk 1 above)
Comment (harryw) [inline features/ignore_configuration_files_in_bundled_gems.feature:11]: Yes it is. And yes, it's case-sensitive and begins with a capital 'T'.

**Pull request merged**